src/dev.config.ts

```diff
@@ -1,15 +1,20 @@
-import baseConfigFactory from './base.config';
+import * as CleanWebpackPlugin from 'clean-webpack-plugin';
 import * as path from 'path';
 import * as webpack from 'webpack';
-import * as CleanWebpackPlugin from 'clean-webpack-plugin';
+import baseConfigFactory from './base.config';
+
+const removeEmpty = (items: any[]) => items.filter(item => item);
 
 function webpackConfig(args: any): webpack.Configuration {
 	const config = baseConfigFactory(args);
 	const { plugins, output } = config;
 	const outputPath = output!.path as string;
 	const location = path.join(outputPath, 'dev');
 
-	config.plugins = [...plugins!, new CleanWebpackPlugin([location], { root: outputPath, verbose: false })];
+	config.plugins = removeEmpty([
+		...plugins!,
+		(args.target !== 'lib' || args.clean) && new CleanWebpackPlugin([location], { root: outputPath, verbose: false })
```

Comment:
I can't see where this `args.clean` is set, now that we are building the `dev` and `dist` separately do we actually need thie different behaviour anymore?

Reply:
I believe it can be removed (from both here and the dist config), but I'll double-check before blindly doing so.